lib/selections.js

```diff
@@ -53,28 +65,26 @@ export class Selections {
    * @param {SelectionItem & NotificationItem} newItem
    */
   insert (newItem) {
-    const { from, to, notify, isStreamSelection, ...restItem } = newItem
-    if (from > to) {
+    if (newItem.from > newItem.to) {
       throw new Error('Invalid interval')
     }
-    if (!isStreamSelection) {
-      this.remove({ from, to, isStreamSelection })
-    }
-    this._items.push({ from, to, isStreamSelection, ...restItem })
-    if (notify) this._notifications.push({ from, to, notify })
-  }
-
-  /** @param {MinimalSelectionItem} range */
-  callNotifyInRange (range) {
-    for (const notification of this._notifications) {
-      if (isLowerIntersecting(range, notification) ||
-            isUpperIntersecting(range, notification) ||
-            isInsideExisting(range, notification) ||
-            isCoveringExisting(range, notification)
-      ) {
-        notification.notify()
+    newItem.notify = newItem.notify ?? noop
```

Comment:
don't noop the notifiy, if it doesn't exist, skip it, less function calls

Reply:
Got it. removed the noop, instead the notify functions are called like `s.notify?.()`.